pkg/controller/service/controller.go

```diff
@@ -804,8 +804,7 @@ func (s *Controller) addFinalizer(service *v1.Service) error {
 	updated.ObjectMeta.Finalizers = append(updated.ObjectMeta.Finalizers, servicehelper.LoadBalancerCleanupFinalizer)
 
 	klog.V(2).Infof("Adding finalizer to service %s/%s", updated.Namespace, updated.Name)
-	// TODO(87447) use PatchService from k8s.io/cloud-provider/service/helpers
-	_, err := patch(s.kubeClient.CoreV1(), service, updated)
```

Comment:
Looks like we can delete the `patch.go` and `patch_test.go` now

Reply:
Yes. Shall I do it as a part of this PR or shall I create a seperate PR to delete patch{_test}.go files?

Reply:
This PR please